zerver/models.py

```diff
@@ -417,12 +414,31 @@ def get_realm_emoji_uncached(realm: Realm) -> Dict[Text, Dict[str, Any]]:
     from zerver.lib.emoji import get_emoji_url
     for row in RealmEmoji.objects.filter(realm=realm).select_related('author'):
         author = None
+        if row.author:
+            author = {
+                'id': row.author.id,
+                'email': row.author.email,
+                'full_name': row.author.full_name}
+        d[str(row.id)] = dict(id=str(row.id),
+                              name=row.name,
+                              source_url=get_emoji_url(row.file_name, row.realm_id),
+                              deactivated=row.deactivated,
+                              author=author)
+    return d
+
+def get_active_realm_emoji_uncached(realm: Realm) -> Dict[str, Dict[str, Any]]:
+    d = {}
+    from zerver.lib.emoji import get_emoji_url
+    for row in RealmEmoji.objects.filter(realm=realm,
+                                         deactivated=False).select_related('author'):
+        author = None
         if row.author:
             author = {
                 'id': row.author.id,
                 'email': row.author.email,
                 'full_name': row.author.full_name}
         d[row.name] = dict(id=str(row.id),
+                           name=row.name,
```

Comment:
What's the story with these changes?  Could they be their own prep commit?

Also, I think it'd be worth de-duplicating most of the code in these two functions; the only difference is a small change to the query-set definition now, right?